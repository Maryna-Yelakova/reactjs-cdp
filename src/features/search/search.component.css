.search_block {
	position: relative;
	height: 400px;
	width: 100%;
	background: url(../../assets/images/search-background.jpg) center center no-repeat;
	background-size: cover;
	box-sizing: border-box;
}

.search_block::before {
	content: '';
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	background-image: linear-gradient(to bottom right, #002f4b, #dc4225);
	opacity: .8;
	z-index: 2;
}

.search_content {
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	z-index: 3;
	padding: 40px;
	display: flex;
	flex-direction: column;
	justify-content: space-between;
}

.search_heading {
	font-size: 30px;
	color: #ff4d73;
}

.search_label {
	font-size: 18px;
	color: #fff;
	text-transform: uppercase;
	margin-bottom: 20px;
}

.search_holder {
	position: relative;
	display: block;
	width: 100%;
	margin-bottom: 10px;
	height: 58px;
	line-height: 3;
	overflow: hidden;
  border-bottom: 2px solid #ff4d73;
}

.search_input {
  display: block;
	width: 100%;
	height: 100%;
	margin: 0;
  outline: none;
  border: 0;
	font-size: 18px;
	font-weight: 300;
  color: #fff;
	box-sizing: border-box;
	padding-left: 16px;
  background-color: #000;
}

.search_control {
  display: flex;
  justify-content: space-between;
}

.search_control_button {
  display: inline-block;
  font-size: 16px;
  color: #fff;
  background-color: #ff4d73;
  padding: 10px 45px;
  border: 0;
  outline: none;
  border-radius: 4px;
}

.search_control_button:hover {
  cursor: pointer;
}

.search_control_filter > span {
  font-size: 16px;
  color: #fff;
  margin-right: 10px;
}

.search_control_filter > .search_control_button {
  padding: 6px 12px;
  background-color: #6b6b6b;
  margin-right: 10px;
}
.search_control_filter > .search_control_button:hover {
  background-color: #ff4d73;
}
