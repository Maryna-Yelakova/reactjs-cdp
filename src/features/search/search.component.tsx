import * as React from 'react';
import { withRouter } from 'react-router';
import './search.component.css';

export const Search = withRouter(({history, children, onChange, setFieldName, field}) => {
  let  searchInput;
  return (
          <div className="search_block">
            <div className="search_content">
              <h1 className="search_heading">Some Movies-box</h1>
              <div>
                <p className="search_label">Find your movie</p>
                <div className="search_holder">
                  <input type="text" className="search_input" id="query" ref={el => searchInput = el} onChange={onChange} />
                </div>
                <div className="search_control">
                  <div className="search_control_filter">
                    <span>Search by:</span>
                    <button className="search_control_button" onClick={() => setFieldName('title')}>Title</button>
                    <button className="search_control_button" onClick={() => setFieldName('genre')}>Genre</button>
                  </div>
                  <button className="search_control_button" onClick={() => history.push(`/search/${searchInput.value}`)}>SEARCH</button>

                </div>
              </div>
            </div>
          </div>
          );
})
