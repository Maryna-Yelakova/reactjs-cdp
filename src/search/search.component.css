.search_block {
  background-color: grey;
  width: 700px;
  height:200px;
  margin: 10px 40px;
}
.search_input {
  display: block;
  position: relative;
  float: right;
  width : 120px;
}
