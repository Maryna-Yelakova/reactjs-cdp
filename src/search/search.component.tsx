import * as React from 'react';
import './search.component.css';

export const Search = ({query, onChange}) => {
  return (
          <div className="search_block">
               <input type="text" className="search_input" value={query} onChange={onChange}/>
          </div>
          );
}
