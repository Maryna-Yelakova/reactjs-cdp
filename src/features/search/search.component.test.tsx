import * as  React from 'react';
import { shallow, mount, render } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { Search } from './search.component';


describe('Search', () => {
 it('should render correctly', () => {
   const changeQuery = jest.fn();
   const component = shallow(<Search query="MockQuery" onChange={changeQuery} />);

   expect(component).toBeDefined();
   expect(shallowToJson(component)).toMatchSnapshot();
 });

 it('handle search', () => {
   const changeQuery = jest.fn();
   const component = mount(<Search query="MockQuery" onChange={changeQuery} />);

  const input = component.find('input');
  input.simulate('change');

  expect(changeQuery).toHaveBeenCalled(); // TODO test onChange
 })
});
