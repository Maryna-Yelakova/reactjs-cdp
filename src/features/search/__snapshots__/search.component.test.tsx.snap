// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div
  className="search_block"
>
  <div
    className="search_content"
  >
    <h1
      className="search_heading"
    >
      Some Movies-box
    </h1>
    <div>
      <p
        className="search_label"
      >
        Find your movie
      </p>
      <div
        className="search_holder"
      >
        <input
          className="search_input"
          id="query"
          onChange={[MockFunction]}
          type="text"
          value="MockQuery"
        />
      </div>
      <div
        className="search_control"
      >
        <div
          className="search_control_filter"
        >
          <span>
            Search by:
          </span>
          <button
            className="search_control_button"
          >
            Title
          </button>
          <button
            className="search_control_button"
          >
            Genre
          </button>
        </div>
        <button
          className="search_control_button"
        >
          SEARCH
        </button>
      </div>
    </div>
  </div>
</div>
`;
